lib/fuzzer/InputPoolPosix.zig

```diff
@@ -0,0 +1,168 @@
+// The inputs are densly packed inside a file that all fuzzing processes have
+// mapped into memory. To know where inputs start and end, there is a second
+// file mapped by all fuzzing processes that stores the end index of every
+// stored string and some metadata (see the meta field).
+//
+// When a process finds a new good input, it (under a lock) appends to both of
+// these files. All other processes can immediately start working on this input
+//
+// We never shrink the corpus when fuzzing. One has to stop fuzzing and run a
+// separate program. That is because erasing strings would require complicated
+// synchronization between fuzzing processes. Finding good inputs seems to be
+// rare enough for the input pool to never get too large.
+
+const std = @import("std");
+const assert = std.debug.assert;
+const util = @import("util.zig");
+const check = util.check;
+const MemoryMappedList = @import("memory_mapped_list.zig").MemoryMappedList;
+
+/// maximum 2GiB of input data should be enough. 32th bit is delete flag
+pub const Index = u31;
```

Comment:
```suggestion
pub const Flags = packed struct (u32) {
    index: Index,
    delete: bool,
    
    pub const Index = enum(u31) {
        /// This tag is used below in the line with
        /// `MemoryMappedList(u8).init(buffer_file, std.math.maxInt(Index))`
        /// but I don't see what the special value means. Replace these doc comments
        /// and field name with a more descriptive one.
        /// Or, if you did not mean for this to be a special value, delete this field and
        /// initialize `index` with `undefined`.
        special_value = std.math.maxInt(u31),
        _,
    };
    
    pub const empty: Flags = .{
        .index = .special_value,
        .delete = true,
    };
};
```

I promise you this type safety will come in handy.

Then you can elsewhere do this:

```diff
-    const buffer = MemoryMappedList(u8).init(buffer_file, std.math.maxInt(Index));
+    const buffer = MemoryMappedList(u8).init(buffer_file, Flags.empty);
```

Instead of `deleteMask`, use `flags.delete` etc.

Reply:
> special_value = std.math.maxInt(u31),

That is not a special value! :D I actually want to allocate 2GiB of virtual memory. It simplifies the problem of reallocating our mmaped file when some *other* process appended to the file.

This way all the process needs to do is read `len` (start of meta file) and access the already allocated buffer. the process that incremented `len` already called ftruncate on the file so all other processes can access their allocated buffer without doing anything at all.

Otherwise there would need to be some logic for other processes to store the allocation size and mremap it when  grows too much and such and such.

Reply:
> Instead of deleteMask, use flags.delete etc

+1